python/stretch_factory/firmware_updater.py

```diff
@@ -621,22 +622,64 @@ def set_target_from_install_version(self):
                     click.secho('---------- %s [%s]-----------' % (
                     device_name.upper(), str(self.fw_installed.get_version(device_name))), fg="blue", bold=True)
                     default_id = 0
-                    for i in range(len(vs)):
+
+                    ## Checks to hw id to ensure that user can not downgrade fw to far
+                    for f_limit in range(len(vs)):  
+                        if self.fw_installed.get_hw_id(device_name) >= 3:
```

Comment:
We can get rid of this if statement here. Instead introduce a class variable, `self.min_allowed_fw_version`, which is a dictionary mapping hardware IDs to minimum allowed fw versions. Something like:

```python
self.min_allowed_fw_version = {
    0: '0.3.1p2',
    1: '0.3.1p2',
    2: '0.3.1p2',
    3: '0.7.0p5',
    4: '0.7.0p5',
}
```

Then the if/else block here is simplified to

```python
for f_limit in range(len(vs)): 
    fw_limit = self.min_allowed_fw_version.get(self.fw_installed.get_hw_id(device_name), None)
    if fw_limit is None:
        raise ValueError(f'unable to determine fw_limit from hardware ID {blah}') # exit out with error message asking user to contact Hello Robot Support
    ...
```

Notice the catch case for if the hardware ID isn't one in our dictionary. This might happen if we shipped new hardware of ID 5 and forgot to update the dictionary above. After the user contacted us, we would update it and ship a new version of Stretch Factory.